provisioner/terraform/testdata/generate.sh

```diff
@@ -3,37 +3,73 @@
 set -euo pipefail
 cd "$(dirname "${BASH_SOURCE[0]}")"
 
-for d in */; do
-	pushd "$d"
+generate() {
+	local name="$1"
+
+	echo "=== BEGIN: $name"
+	terraform init -upgrade &&
+		terraform plan -out terraform.tfplan &&
+		terraform show -json ./terraform.tfplan | jq >"$name".tfplan.json &&
+		terraform graph -type=plan >"$name".tfplan.dot &&
+		rm terraform.tfplan &&
+		terraform apply -auto-approve &&
+		terraform show -json ./terraform.tfstate | jq >"$name".tfstate.json &&
+		rm terraform.tfstate &&
+		terraform graph -type=plan >"$name".tfstate.dot
```

Comment:
Are the `&&`'s necessary if we have `-e` set?

Reply:
It shouldn't be, but I was having trouble getting this working without them. I might've had some other issue before that caused it though so I can retry.

Reply:
On the other hand, printing "end" is dependent on the chaining, which makes it easier to discern what output belongs where.